src/host.cpp

```diff
@@ -838,6 +923,34 @@ namespace rhost {
             unblock_message_loop();
         }
 
+        void handle_binary_message(ws_connection_type::message_ptr msg) {
+            if (binary_message_queue.size() != 1) {
+                fatal_error("Unexpected incoming client message - host was expecting a single pending binary message.");
+            }
+
+            message incoming = binary_message_queue.front();
+            const auto& raw = msg->get_raw_payload();
+            rhost::util::blob_slice slice(raw.data(), raw.data() + raw.size());
+            incoming.blob.push_back(std::move(slice));
```

Comment:
This will be even faster and shorter with `emplace_back` to construct the slice directly inside the vector.